src/policy/policy.h

```diff
@@ -35,6 +35,8 @@ static constexpr unsigned int MAX_STANDARD_TX_SIGOPS_COST{MAX_BLOCK_SIGOPS_COST/
 static constexpr unsigned int DEFAULT_INCREMENTAL_RELAY_FEE{1000};
 /** Default for -bytespersigop */
 static constexpr unsigned int DEFAULT_BYTES_PER_SIGOP{20};
+/** Default for -permitbarepubkey */
+static constexpr bool DEFAULT_PERMIT_BAREPUBKEY{true};
```

Comment:
Concept ACK because default policy is unchanged.

> I don't care if this gets merged, personally **I don't see any benefit of having this option** (or -permitbaremultisig) and I would object to changing the default policy in this regard (similar to my objections to https://github.com/bitcoin/bitcoin/pull/28217).

It might be useful for testing and a few other use cases.

> Concept ACK, it's great to give users the option of filtering out potentially abusive txs

The only potential abuse I foresee in the future is trying to make it default in another pull request without rationale, agreement from reviewers, etc.